compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1074,7 +1075,15 @@ public void visit(BLangArrayType arrayTypeNode) {
                         continue;
                     }
 
-                    int length = Integer.parseInt(sizeConstSymbol.type.toString());
+                    int length;
+                    long lengthCheck = Long.parseLong(sizeConstSymbol.type.toString());
```

Comment:
Thinking out loud, one other approach we can take to avoid duplicating the logic is to maybe maintain the size maybe as a string at node transformer level and do the parsing and checking here.

Reply:
I created an issue for this. #32034